oci/utils_unix_test.go

```diff
@@ -37,6 +39,70 @@ func cleanupTest() {
 	usernsRunningInUserNS = userns.RunningInUserNS
 }
 
+// Should not error on symlinked device
+func TestHostDeviceIsAbsoluteSymlink(t *testing.T) {
+	zero := "/dev/zero"
+	if _, err := os.Stat(zero); err != nil {
+		t.Skip("Host does not have /dev/zero")
+	}
+
+	dir := t.TempDir()
+	symZero := filepath.Join(dir, "zero")
+
+	// Create symlink to /dev/zero
+	if err := os.Symlink(zero, symZero); err != nil {
+		t.Fatalf("unexpected error creating symlink: %q", err)
+	}
+
+	// The symlink won't be in /dev, so we must manually add it to devices
+	device, err := DeviceFromPath(symZero)
+	if err != nil {
+		t.Fatalf("error getting symlinked device from path: %q", err)
+	}
+
+	evaluateDevice(*device, t)
+}
+
+func TestHostDeviceIsRelativeSymlink(t *testing.T) {
+	zero := "/dev/zero"
+	relativeZero := "./zero"
+	if _, err := os.Stat(zero); err != nil {
+		t.Skip("Host does not have /dev/zero")
+	}
+
+	// Create symlink to /dev/zero
+	if err := os.Symlink(zero, relativeZero); err != nil {
+		t.Fatalf("unexpected error creating symlink: %q", err)
+	}
+	defer os.Remove(relativeZero)
+
+	// The symlink won't be in /dev, so we must manually add it to devices
+	device, err := DeviceFromPath("./zero")
+	if err != nil {
+		t.Errorf("unexpected error getting symlink from path: %q", err)
+	}
+
+	evaluateDevice(*device, t)
+}
+
+// Should fail; symlink should point to another device, not a regular file
+func TestHostDeviceSymlinkToRegularFile(t *testing.T) {
+	// Create temp dir for creating a symlink
+	dir := t.TempDir()
+	fakeDevice := filepath.Join(dir, "fakeDevice")
+	symFakeDevice := filepath.Join(dir, "sym")
+
+	// Create symlink to /dev/zero
+	if err := os.Symlink(fakeDevice, symFakeDevice); err != nil {
+		t.Fatalf("unexpected error creating symlink: %q", err)
+	}
+	// The symlink won't be in /dev, so we must manually add it to devices
+	_, err := DeviceFromPath(symFakeDevice)
+	if err == nil {
+		t.Error("should fail")
+	}
```

Comment:
Do you have a specific error you'd expect here?

Reply:
updated to check for os.ErrNotExist